src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IncentiveReviewsService.kt

```diff
@@ -56,8 +68,9 @@ class IncentiveReviewsService(
           positiveBehaviours = positiveCaseNotesInLast3Months[it.prisonerNumber]?.totalCaseNotes ?: 0,
           negativeBehaviours = negativeCaseNotesInLast3Months[it.prisonerNumber]?.totalCaseNotes ?: 0,
           acctOpenStatus = it.acctOpen,
+          nextReviewDate = nextReviewDates[it.bookingId.toLong()]!!.nextReviewDate,
```

Comment:
isn't the `!!` going to throw a null pointer exception? I wouldn't think the value for any `bookingId` shouldn't be missing or we would get a `ListOfDataNotFoundException` above right?

Is it because `[]` returns `T?` and Kotlin is complaining unless you use `!!`? Maybe it doesn't matter in this case as we know we should never get a map with some of the `bookingId` missing right?

Reply:
> Is it because [] returns T? and Kotlin is complaining unless you use !!? Maybe it doesn't matter in this case as we know we should never get a map with some of the bookingId missing right?

Exactly, on both points. I'm equally not a huge fan of `!!` but we've protected against this handling `ListOfDataNotFoundException`